--- opencv_contrib-3.0.0\modules\line_descriptor\src\bitops.hpp
+++ opencv_contrib-3.0.0\modules\line_descriptor\src\bitops.hpp
@@ -101,7 +101,7 @@
   UINT64 temp = 0x0;
   int nbits = 0;
   int nbyte = 0;
-  UINT64 mask = b == 64 ? 0xFFFFFFFFFFFFFFFFLLU : ( ( UINT64_1 << b ) - UINT64_1 );
+  UINT64 mask = b == 64 ? 0xFFFFFFFFFFFFFFFFULL : ( ( UINT64_1 << b ) - UINT64_1 );
 
   for ( int i = 0; i < m; i++ )
   {

