--- opencv_contrib-3.0.0/modules/line_descriptor/src/LSDDetector.cpp
+++ opencv_contrib-3.0.0/modules/line_descriptor/src/LSDDetector.cpp
@@ -179,7 +179,7 @@
       kl.sPointInOctaveY = (float) extremes[1];
       kl.ePointInOctaveX = (float) extremes[2];
       kl.ePointInOctaveY = (float) extremes[3];
-      kl.lineLength = (float) sqrt( pow( extremes[0] - extremes[2], 2 ) + pow( extremes[1] - extremes[3], 2 ) );
+      kl.lineLength = (float) sqrt( pow( (float) extremes[0] - extremes[2], 2 ) + pow( (float) extremes[1] - extremes[3], 2 ) );
 
       /* compute number of pixels covered by line */
       LineIterator li( gaussianPyrs[j], Point( extremes[0], extremes[1] ), Point( extremes[2], extremes[3] ) );
